File: frontend/src/types/teamCosts.ts
/**
 * Team costs related types and interfaces
 */

export interface TeamCostItem {
  team_name: string;
  total_cost: number;
  percentage: number;
  resource_count: number;
  avg_cost_per_resource: number;
  color: string;
}

export interface TeamCostsResponse {
  success: boolean;
  data: TeamCostItem[];
  total_cost: number;
  period: string;
  team_count: number;
  last_updated: string;
  metadata: {
    query_params?: {
      time_period?: string;
      provider?: string;
      account_id?: string;
      environment?: string;
      cost_center?: string;
    };
    data_source?: string;
    aggregation_method?: string;
    message?: string;
  };
}

export interface TeamCostsRequest {
  time_period?: string;
  provider?: string;
  account_id?: string;
  environment?: string;
  cost_center?: string;
  limit?: number;
}

export interface TeamDetails {
  team_name: string;
  period: string;
  services: Array<{
    provider: string;
    service_name: string;
    cost: number;
    resource_count: number;
  }>;
  total_services: number;
}

// Chart data formats for visualization
export interface TeamCostChartData {
  name: string;
  value: number;
  percentage: number;
  color: string;
  fullName?: string;
}

export interface TeamCostsSummary {
  totalTeams: number;
  totalCost: number;
  highestTeam: {
    name: string;
    cost: number;
    percentage: number;
  };
  lowestTeam: {
    name: string;
    cost: number;
    percentage: number;
  };
}

File: backend/app/schemas/team_costs.py
"""
Schemas for team costs API endpoints
"""
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class TeamCostItem(BaseModel):
    """Individual team cost item"""
    team_name: str = Field(..., description="Name of the team")
    total_cost: float = Field(..., description="Total cost for the team")
    percentage: float = Field(..., description="Percentage of total costs")
    resource_count: int = Field(..., description="Number of resources for this team")
    avg_cost_per_resource: float = Field(..., description="Average cost per resource")
    color: str = Field(..., description="Color for chart visualization")


class TeamCostsResponse(BaseModel):
    """Response model for team costs endpoint"""
    success: bool = Field(True, description="Whether the request was successful")
    data: List[TeamCostItem] = Field(..., description="List of team cost items")
    total_cost: float = Field(..., description="Total cost across all teams")
    period: str = Field(..., description="Time period for the data")
    team_count: int = Field(..., description="Total number of teams")
    last_updated: datetime = Field(..., description="When the data was last updated")
    metadata: dict = Field(default_factory=dict, description="Additional metadata")


class TeamCostsRequest(BaseModel):
    """Request parameters for team costs"""
    time_period: str = Field("30d", description="Time period (e.g., '30d', '7d', '90d')")
    provider: Optional[str] = Field(None, description="Filter by cloud provider")
    account_id: Optional[str] = Field(None, description="Filter by account ID")
    environment: Optional[str] = Field(None, description="Filter by environment")
    cost_center: Optional[str] = Field(None, description="Filter by cost center")
    limit: int = Field(10, description="Maximum number of teams to return")
